src/checkers/urlchecker.py

```diff
@@ -92,9 +93,17 @@ async def check(self, external_data: ExternalBase):
         else:
             url = external_data.current_version.url
 
+        strip_query = external_data.checker_data.get("strip-query", False)
+
         version_string = None
 
         try:
+            if strip_query:
+                async with self.session.head(
+                    url, allow_redirects=True, headers=HTTP_CLIENT_HEADERS
+                ) as head:
+                    url = str(head.url.with_query(""))
```

Comment:
Why is it necessary to traverse redirects?

Reply:
Because the initial URL without a query can redirect us to an URL with a query, and we want to make sure it works with the query removed.

Reply:
I don't quite follow, but I'll take your word for it :)

Reply:
Well, for example, `http://example.com/download/file.tar.gz` can redirect to `http://cdn.example.com/pub/file.tar.gz?id=key`. Since we're stripping the query from the final url, we should check that `http://cdn.example.com/pub/file.tar.gz` works without `id=key`. Otherwise, we can end up submitting and update with broken URL (in case it doesn't work without query).